org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/MavenBuildSupportTest.java

```diff
@@ -177,6 +179,42 @@ public void testDownloadSources() throws Exception {
 		assertNotNull(source);
 	}
 
+	@Test
+	public void testUpdateSnapshots() throws Exception {
```

Comment:
Does this test fail without the fix?

Reply:
No, it doesn't fail. We are testing whether a dependency has actually been added.

Reply:
if it doesn't fail with or without the fix, then the test proves nothing

Reply:
The new preference doesn't add any functionality. It just enhances performance. We are testing whether it breaks the update maven project action.

Reply:
@fbricon Do you want me to remove this test?